### ISSUE TITLE
UnnecessaryFullyQualifiedName false when inner class is instantiated from a instance variable of the parent class

### ISSUE THREAD

**Issue by atulgpt** [detekt/detekt#9142]
## Expected Behavior
```kt
@Test
fun `does not report when inner class instance is created from the outer class instance`() {
    val code = """
        class Test {
            inner class A
        }
        
        fun main(test: Test) {
            val a = Test().A()
            val b = test.A()
        }
    """.trimIndent()

    assertThat(subject.lint(code)).isEmpty()
}
```

Above TC should pass

## Observed Behavior
Above TC is failing

## Steps to Reproduce
Detekt should be run in main branch or alphas release where rule `UnnecessaryFullyQualifiedName`

**Comment by thejamesgore** [comment]
Taking a look at this one

**Comment by hudson155** [comment]
Just came here to report the same issue — thanks for looking into it :)

```kotlin
@Single
class ContextFactory(
  private val historyFactory: HistoryFactory,
  private val modelFactory: ModelFactory,
  private val configureContext: ConfigureContext,
) {
  fun create(conversation: ConversationModel): Context =
    context {
      userId = conversation.userId
      json = osirisJson
      history = historyFactory.History(conversation.id) // THE FALSE POSITIVE IS HERE
      configureContext(modelFactory)
    }
}
```
